modules/fedimint-ln-server/src/lib.rs

```diff
@@ -855,16 +840,29 @@ impl Lightning {
         Ok(Lightning { cfg, btc_rpc })
     }
 
-    /// Returns the median block height if enough peers have contributed
-    pub async fn consensus_height(&self, dbtx: &mut ModuleDatabaseTransaction<'_>) -> Option<u64> {
-        let peer_heights = dbtx.get_value(&BlockHeightKey).await.unwrap_or_default();
-        if peer_heights.len() < self.cfg.consensus.threshold() {
-            return None;
+    pub async fn block_height(&self) -> u64 {
+        self.btc_rpc
+            .get_block_height()
+            .await
+            .expect("bitcoind rpc failed")
+    }
+
+    pub async fn consensus_block_height(&self, dbtx: &mut ModuleDatabaseTransaction<'_>) -> u64 {
+        let peer_count = self.cfg.consensus.threshold_pub_keys.coefficients().len();
+
+        let mut heights = vec![];
+
+        for index in 0..peer_count {
+            heights.push(
+                dbtx.get_value(&BlockHeightVoteKey(PeerId::from(index as u16)))
+                    .await
+                    .unwrap_or(0),
+            )
         }
```

Comment:
Could this be a single call to `dbtx.find_by_prefix` instead?

Reply:
Done